overlord/snapstate/handlers.go

```diff
@@ -115,6 +115,88 @@ func snapSetupAndState(t *state.Task) (*SnapSetup, *SnapState, error) {
 
 */
 
+const defaultCoreSnapName = "core"
+const defaultBaseSnapsChannel = "stable"
+
+func (m *SnapManager) doPrerequisites(t *state.Task, _ *tomb.Tomb) error {
+	// the state lock is not enough to ensure this is not run in
+	// parallel as it is dropped in snapstate.Install() below
+	m.prerequisitesLock.Lock()
+	defer m.prerequisitesLock.Unlock()
```

Comment:
we tend to use SetBlocked callbacks to serialize tasks, not locks, so they don't even start